cpp/src/table/legacy/table.cpp

```diff
@@ -27,14 +27,17 @@
 namespace cudf {
 
 table::table(std::vector<gdf_column*> const& cols) : _columns{cols} {
-  CUDF_EXPECTS(nullptr != cols[0], "Null input column");
-  gdf_size_type num_rows = cols[0]->size;
-
-  std::for_each(_columns.begin(), _columns.end(),
-                [this, num_rows](gdf_column* col) {
-                  CUDF_EXPECTS(nullptr != col, "Null input column");
-                  CUDF_EXPECTS(num_rows == col->size, "Column size mismatch");
-                });
+  if (cols.size() != 0)
```

Comment:
You'll want to update your PR with the latest 0.10 branch which added throwing an exception if you try and construct a table with 0 columns.